### DISCUSSION TITLE
hyper parameters not restored while resuming training

### DISCUSSION THREAD

**Discussion by Haojia521** [Lightning-AI/pytorch-lightning#12639] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I call `save_hyperparameters()` in `__init__()`, and all hyper parameters sent to PL model are saved to checkpoint file. However, when i resume training from a checkpoint(call `trainer.fit(..., ckpt_path=checkpoint_file_path)`), the hyper parameters are not restored from checkpoint file and all of them keep initial values.

**Comment by rohitgr7** [comment, marked as answer]
hyperparameters are not restored by default because it allows users to update them if they want, using the checkpoint, while resuming.

you can do this:
```py
model = LitModel.load_from_checkpoint(checkpoint_file_path)
trainer.fit(model, ..., ckpt_path=checkpoint_file_path)
```

**Reply by Haojia521** [reply]
ok, it works fine. Thanks very much for your advice.
